Sources/HummingbirdTLS/TLSChannel+ConfigReader.swift

```diff
@@ -0,0 +1,62 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2025 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if ExperimentalConfiguration
+
+public import Configuration
+
+@available(macOS 15, iOS 18, macCatalyst 18, tvOS 18, visionOS 2, *)
+extension TLSChannelConfiguration {
+    public struct TLSConfigError: Error {
+        private enum Internal {
+            case missingCertificateChain
+            case missingPrivateKey
+        }
+        private let value: Internal
+
+        public static var missingCertificateChain: Self { .init(value: .missingCertificateChain) }
+        public static var missingPrivateKey: Self { .init(value: .missingPrivateKey) }
+    }
+
+    /// Initialize a TLSChannelConfiguration from a ConfigReader
+    ///
+    /// - Configuration Keys
+    ///   - `tls.certificate.chain` (string, required): TLS certificate chain in PEM format
+    ///   - `tls.private.key` (string required): TLS private key in PEM format
+    ///   - `tls.trust.roots` (string optional): TLS trust roots in PEM format
+    ///
+    /// - Parameter reader: ConfigReader
+    /// - Throws: TLSConfigError if "tls.certificate.chain" or "tls.private.key" values are missing
+    public init(reader: ConfigReader) throws {
+        guard let certificateChainPEM = reader.string(forKey: "tls.certificate.chain") else {
```

Comment:
```suggestion
        guard let certificateChainPEM = reader.string(forKey: "tls.certificateChain") else {
```

`certificate` is not a namespace. If we want to namespace it, it's more likely your "local" certificate: `local.certificate` or `local.privateKey` vs. `remote.certificate` for certificate pinning.